### PR TITLE
Version 0.8.1

### PR DESCRIPTION
**Pull request by lassepe**
The draft PR for the version 0.8.1 release.

### CODE REVIEW

**Comment by zsunberg** [comment]
@MaximeBouton and @lassepe, can you guys make a quick review to check that this makes sense this since you commented a lot in #274 ?

**Comment by zsunberg** [comment]
@lassepe I am not sure why I can't request your review

File: src/errors.jl [hunk 1]
```diff
@@ -86,6 +86,14 @@ function gen_analysis(io, ex::DistributionNotImplemented)
     end
 end
 
+function try_show_method_candidates(io, args...)
+    try
+        Base.show_method_candidates(io, args...) # this isn't exported, so it might break
+    catch ex
+        @debug("Unable to show method candidates:\n$(sprint(showerror, ex))")
```
Comment (lassepe) [inline src/errors.jl]: Maybe this should say "Please submit an issue"?

**Comment by lassepe** [comment]
As we are at it: In `test/test_requirements.jl` there is the glorious print "There should be no warnings or requirements output below this point!". However, there are warning generated in `test/test_deprecated_generative.jl`. We should either change the order in which things are tested or remove the comment.

**Comment by lassepe** [comment]
I just checked whether we can simply pattern-match against all warnings in `test/test_deprecated_generative.jl` but somehow `@test_logs` fails to catch the warning from `gen` (maybe because it is `@generated`?).

**Comment by lassepe** [comment]
There is also [JuiliaIO/Suppressor.jl](https://github.com/JuliaIO/Suppressor.jl).

File: docs/src/generative.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ The *generative* interface consists of two functions:
 The generative interface is typically used when it is easier to return sampled states and observations rather than explicit distributions as in the [Explicit interface](@ref explicit_doc).
 This type of model is often referred to as a "black-box" model.
 
+In some special cases (e.g. reinforcement learning), an initial observation is needed before any actions are taken. In this case, the [`initialobs`](@ref) function will be used.
```
[the remark this answers is not in the document]
Comment (zsunberg) [reply docs/src/generative.md]: done

File: src/errors.jl (hunk 1 above)
Comment (zsunberg) [reply src/errors.jl]: done

**Comment by lassepe** [comment]
I can't approve because I opened the PR but I think this is good!

**Comment by zsunberg** [comment]
Ok, thanks - I am trying to get rid of all the warnings and automate checking them with Suppressor. Your idealism has inspired me, haha

File: src/errors.jl [hunk 3]
```diff
@@ -3,7 +3,7 @@ struct DistributionNotImplemented <: Exception
     gen_firstarg::Type
     func::Function
     modeltype::Type
-    dep_argtypes::NamedTuple
+    dep_argtypes::Union{AbstractVector, Core.SimpleVector}
```
Comment (MaximeBouton) [inline src/errors.jl]: what is `Core.SimpleVector`?

**Comment by zsunberg** [comment]
NVM getting rid of the warning printing is too hard and not a huge benefit

**Comment by lassepe** [comment]
> NVM getting rid of the warning printing is too hard and not a huge benefit

You should add a comment "I have tried suppressing these warnings for <counter> minutes. If you keep working on this, please increase this counter."

File: src/errors.jl (hunk 3 above)
Comment (zsunberg) [reply src/errors.jl]: It is used in some type parameter things. I think it is a simpler vector version for use in the compiler before `Array` is compiled

Comment (lassepe) [reply src/errors.jl]: Maybe in this case it would be better to just duck type it and hope that it quacks like a vector.

Comment (zsunberg) [reply src/errors.jl]: I just manually converted it to a Vector. (It does seem to quack exactly like a Vector, but it is not an AbstractVector and there is no `convert` to an AbstractVector)

**Pull request merged**